File: Proj4b/Questão 33/Main.java
import java.util.Scanner;

public class Main {

    public static double sqrtApprox(double A, int n) {
        double x = 1.0; // inicializa x1 como 1

        for (int i = 0; i < n; i++) {
            x = (x + A / x) / 2.0;
        }

        return x;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o valor de A (positivo): ");
        double A = scanner.nextDouble();

        System.out.print("Digite o número de iterações n (inteiro e positivo): ");
        int n = scanner.nextInt();

        if (A <= 0 || n <= 0) {
            System.out.println("Valores inválidos para A ou n.");
        } else {
            double result = sqrtApprox(A, n);
            System.out.println("A aproximação da raiz quadrada de " + A + " é: " + result);
        }

        scanner.close();
    }
}

File: Proj4b/Questão 33/main.c
#include <stdio.h>

// Função para calcular a aproximação da raiz quadrada
double sqrtApprox(double A, int n) {
    double x = 1.0; // inicializa x1 como 1

    for (int i = 0; i < n; i++) {
        x = (x + A / x) / 2.0;
    }

    return x;
}

int main() {
    double A;
    int n;

    printf("Digite o valor de A (positivo): ");
    scanf("%lf", &A);

    printf("Digite o número de iterações n (inteiro e positivo): ");
    scanf("%d", &n);

    if (A <= 0 || n <= 0) {
        printf("Valores inválidos para A ou n.\n");
    } else {
        double result = sqrtApprox(A, n);
        printf("A aproximação da raiz quadrada de %lf é: %lf\n", A, result);
    }

    return 0;
}
